calico_containers/pycalico/datastore.py

```diff
@@ -291,9 +292,25 @@ def ensure_global_config(self):
         # Disable felix status reporting, which consumes etcd write bandwidth.
         self._write_global_config(FELIX_REPORTING_INTERVAL_PATH, "0")
 
+        # Create our Cluster GUID (if it does not already exist).
+        self._create_cluster_guid(CLUSTER_GUID_PATH)
+
         # We are always ready.
         self.etcd_client.write(CALICO_V_PATH + "/Ready", "true")
 
+    def _create_cluster_guid(self, key):
+        """
+        Create a globally unique cluster GUID.  Write it idempotently into the
+        datastore, if an exist cluster GUID doesn not already  exist.
+        The prevExist=False ensures an idempotent create operation.
+        """
+        try:
+            self.etcd_client.read(key)
+        except EtcdKeyNotFound:
+            UUID = uuid.uuid4()
+            uuid_string = UUID.get_hex()
+            self.etcd_client.write(key, uuid_string, prevExist=False)
```

Comment:
If the CaS fails, `write()` will raise an exception (I think it'll be `EtcdCompareFailed`), which you should catch and ignore.